### PR TITLE
Initial Tensorflow work PR: Basic TF support

### PR DESCRIPTION
**Pull request by ethantang95**
Includes

- using tensorflow with digits
- necessary example now have tensorflow version
- tensorflow documentation
- Travis build scripts for tensorflow

Please review @lukeyeager, @gheinrich, @TimZaman

### CODE REVIEW

**Comment by gheinrich** [comment]
Hi Ethan, can you integrate the changes in a way that preserves authorship:
- initial integration (by @tzaman who did most of the work)
- misc bug fixes and GAN example (by @gheinrich)
- your changes

**Comment by TimZaman** [comment]
Like greg says, I'm afraid you oversquashed this indeed!

File: scripts/travis/pypi-upload.sh [hunk 1]
```diff
@@ -15,3 +15,5 @@ username = luke.yeager
 password = ${PYPI_PASSWORD}
 EOF
 twine upload -r pypi dist/*
+
+echo "pypi-upload.sh finished"
```
Comment (lukeyeager) [inline scripts/travis/pypi-upload.sh]: You can remove this debugging code

File: requirements.txt [hunk 2]
```diff
@@ -1,5 +1,5 @@
 Pillow>=2.3.0,<=3.1.2
-numpy>=1.8.1,<=1.11.0
+numpy>1.8.1,<=1.11.0
```
Comment (lukeyeager) [inline requirements.txt]: Why this change? Looks like this would break our 14.04 packaging:
https://packages.ubuntu.com/trusty/python-numpy

**Comment by ethantang95** [comment]
Closing it to fix rebase problems
